File: src/main/java/com/marand/auditor/saver/AuditService.java
package com.marand.auditor.saver;

import java.util.Collection;

import com.marand.auditor.dto.AuditInfo;
import org.springframework.stereotype.Service;

/**
 * @author Nejc Korasa
 */

@SuppressWarnings("unused")
@Service
public interface AuditService
{
  void audit(final AuditInfo auditInfo) throws Exception;

  void recover(Exception ex, AuditInfo auditInfo) throws Exception;

  void audit(final Collection<AuditInfo> auditInfos) throws Exception;
}

File: src/main/kotlin/com/marand/auditor/AuditService.kt
package com.marand.auditor

import com.marand.auditor.db.AuditInfoRepository
import com.marand.auditor.db.model.AuditInfoEntity
import com.marand.auditor.dto.AuditInfo
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

/**
 * @author Nejc Korasa
 */

@Service
open class AuditService(private val auditInfoRepository: AuditInfoRepository) {

    @Transactional
    open fun audit(auditInfo: AuditInfo): AuditInfoEntity = auditInfoRepository.save(AuditInfoMapper.fromDto(auditInfo))
}